src/ChargeDrift/ChargeDrift.jl

```diff
@@ -66,11 +66,12 @@ function modulate_driftvector(sv::CartesianVector{T}, pt::CartesianPoint{T}, vdv
 end
 modulate_driftvector(sv::CartesianVector{T}, pt::CartesianPoint{T}, vdv::Missing) where {T} = sv
 
-@inline function _is_next_point_in_det(pt_car::CartesianPoint{T}, pt_cyl::CylindricalPoint{T}, det::SolidStateDetector{T}, point_types::PointTypes{T, 3, Cylindrical})::Bool where {T <: SSDFloat}
+@inline function _is_next_point_in_det(pt::CartesianPoint{T}, det::SolidStateDetector{T}, point_types::PointTypes{T, 3, Cylindrical})::Bool where {T <: SSDFloat}
+    pt_cyl::CylindricalPoint{T} = CylindricalPoint(pt)
     pt_cyl in point_types || pt_cyl in det
 end
-@inline function _is_next_point_in_det(pt_car::CartesianPoint{T}, pt_cyl::CylindricalPoint{T}, det::SolidStateDetector{T}, point_types::PointTypes{T, 3, Cartesian})::Bool where {T <: SSDFloat}
-    pt_car in point_types || pt_car in det
+@inline function _is_next_point_in_det(pt::CartesianPoint{T}, det::SolidStateDetector{T}, point_types::PointTypes{T, 3, Cartesian})::Bool where {T <: SSDFloat}
+    pt in point_types || pt in det
 end

```

Comment:
There might be cases at surfaces where `pt` is just outside the detector but marked as inside as it still lies in the volume of a grid point which `PointType` is marked as semiconductor. 
`PointType` informations are stored bitwise. I think we have some bits left. 
We could add a flag indicating whether all neighboured grid points are also marked as semiconductor points. 
Maybe also including diagonal neighbours due to tilted surfaces.

Reply:
This, in fact, leads to problems in some cases, as then then `Line` connects a point outside (but marked as "inside") and a point outside (marked as "outside") that does not intersect with the detector surface. I'll look into this and make this PR a draft for now.